benchmarks/auth/nodejs/server.js

```diff
@@ -0,0 +1,127 @@
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+// SOFTWARE.
+
+// Include process module
+const process = require('process');
+const GRPC_PORT = process.env.GRPC_PORT || '50051'
+
+
+var PROTO_PATH = __dirname + '/auth.proto';  // '/../proto/auth.proto';
+
+var grpc = require('@grpc/grpc-js');
+var protoLoader = require('@grpc/proto-loader');
+var packageDefinition = protoLoader.loadSync(
+    PROTO_PATH,
+    {keepCase: true,
+     longs: String,
+     enums: String,
+     defaults: true,
+     oneofs: true
+    });
+var auth_proto = grpc.loadPackageDefinition(packageDefinition).auth;
+
+
+
+
+// https://docs.aws.amazon.com/apigateway/latest/developerguide/apigateway-use-lambda-authorizer.html#api-gateway-lambda-authorizer-create
+// A simple token-based authorizer example to demonstrate how to use an authorization token 
+// to allow or deny a request. In this example, the caller named 'user' is allowed to invoke 
+// a request if the client-supplied token value is 'allow'. The caller is not allowed to invoke 
+// the request if the token value is 'deny'. If the token value is 'unauthorized' or an empty
+// string, the authorizer function returns an HTTP 401 status code. For any other token value, 
+// the authorizer returns an HTTP 500 status code. 
+// Note that token values are case-sensitive.
+
+// Help function to generate an IAM policy
+var generatePolicy = function(principalId, effect, resource) {
+  var authResponse = {};
+  
+  authResponse.principalId = principalId;
+  if (effect && resource) {
+      var policyDocument = {};
+      policyDocument.Version = '2012-10-17'; 
+      policyDocument.Statement = [];
+      var statementOne = {};
+      statementOne.Action = 'execute-api:Invoke'; 
+      statementOne.Effect = effect;
+      statementOne.Resource = resource;
+      policyDocument.Statement[0] = statementOne;
+      authResponse.policyDocument = policyDocument;
+  }
+  
+  // Optional output with custom properties of the String, Number or Boolean type.
+  authResponse.context = {
+      "stringKey": "stringval",
+      "numberKey": 123,
+      "booleanKey": true
+  };
+  return authResponse;
+}
+
+/**
+ * Implements the SayHello RPC method.
+ */
+function sayHello(call, callback) {
+  
+  var token = call.request.name;
+  var fakeMethodArn = "arn:aws:execute-api:{regionId}:{accountId}:{apiId}/{stage}/{httpVerb}/[{resource}/[{child-resources}]]";
+  var msg1, ret;
+  switch (token) {
+      case '.f2':
+          ret = generatePolicy('user', 'Allow', fakeMethodArn);
+          msg1 = 'auth.f2';
+          break;
+      default:  // case '.f1':
+          ret = generatePolicy('user', 'Deny', fakeMethodArn);
+          msg1 = 'auth.f1';
+          break;
+      // case 'unauthorized':
```

Comment:
Please implement the example functionality from AWS.
So switch on `allow` and `deny` tokens instead of `.f1` and then we should return the generated policy as string if possible. Can you check if that works?

```
// A simple token-based authorizer example to demonstrate how to use an authorization token 
// to allow or deny a request. In this example, the caller named 'user' is allowed to invoke 
// a request if the client-supplied token value is 'allow'. The caller is not allowed to invoke 
// the request if the token value is 'deny'. If the token value is 'unauthorized' or an empty
// string, the authorizer function returns an HTTP 401 status code. For any other token value, 
// the authorizer returns an HTTP 500 status code. 
// Note that token values are case-sensitive.

exports.handler =  function(event, context, callback) {
    var token = event.authorizationToken;
    switch (token) {
        case 'allow':
            callback(null, generatePolicy('user', 'Allow', event.methodArn));
            break;
        case 'deny':
            callback(null, generatePolicy('user', 'Deny', event.methodArn));
            break;
        case 'unauthorized':
            callback("Unauthorized");   // Return a 401 Unauthorized response
            break;
        default:
            callback("Error: Invalid token"); // Return a 500 Invalid token response
    }
};
```